### PR TITLE
tcti-device: change device tcti to nonblocking mode

### PR DESCRIPTION
**Pull request by tstruk**
Change device tcti to nonblocking mode of operation.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -64,13 +64,22 @@ AC_ARG_ENABLE([esapi],
             [enable_esapi=$enableval],
             [enable_esapi=yes])
 
-
```
Comment (flihp) [inline configure.ac]: whitespace. This is good cleanup but not related to this PR / feature.

File: configure.ac [hunk 2]
```diff
@@ -64,13 +64,22 @@ AC_ARG_ENABLE([esapi],
...
     [enable_esapi="yes"])
 AM_CONDITIONAL(ESAPI, test "x$enable_esapi" = "xyes")
 
+AC_ARG_ENABLE([tcti-device-async],
+    AS_HELP_STRING([--enable-tcti-device-async],
+	           [Enable asynchronus operation on TCTI device
+		    (note: This needs to be supported by the kernel driver). default is no]),
+    [enable_tcti_device_async=$enableval],
+    [enable_tcti_device_async=no])
+AM_CONDITIONAL(TCTI_ASYNC, test "x$enable_tcti_device_async" = "xyes")
```
Comment (flihp) [inline configure.ac]: I don't see a place in the makefile where you use this variable. Unless I missed it, this isn't necessary.

File: src/tss2-tcti/tcti-device.c [hunk 3]
```diff
@@ -118,7 +119,7 @@ tcti_device_receive (
     TSS2_TCTI_DEVICE_CONTEXT *tcti_dev = tcti_device_context_cast (tctiContext);
     TSS2_TCTI_COMMON_CONTEXT *tcti_common = tcti_device_down_cast (tcti_dev);
     TSS2_RC rc = TSS2_RC_SUCCESS;
-    ssize_t  size;
+    ssize_t size = 0;
```
Comment (flihp) [inline src/tss2-tcti/tcti-device.c]: This good cleanup but it doesn't belong in this commit.

File: src/tss2-tcti/tcti-device.c [hunk 4]
```diff
@@ -251,6 +294,7 @@ Tss2_Tcti_Device_Init (
     TSS2_TCTI_DEVICE_CONTEXT *tcti_dev;
     TSS2_TCTI_COMMON_CONTEXT *tcti_common;
     const char *dev_path = conf != NULL ? conf : TCTI_DEVICE_DEFAULT;
+    mode_t mode = O_RDWR;
```
Comment (flihp) [inline src/tss2-tcti/tcti-device.c]: This isn't a mode, it's a flag since it's used in the second parameter: `int open(const char *pathname, int flags, mode_t mode);`. The reason you're not seeing a type error is that the `flags` paramter is an `int` and `mode_t` maps back to `int`. So this variable should be an `int` and it should be called `flags`.

File: configure.ac (hunk 2 above)
Comment (tstruk) [reply configure.ac]: Removed

File: src/tss2-tcti/tcti-device.c (hunk 3 above)
Comment (tstruk) [reply src/tss2-tcti/tcti-device.c]: The size needs to be initialized because the compiler complains otherwise

File: src/tss2-tcti/tcti-device.c (hunk 4 above)
Comment (tstruk) [reply src/tss2-tcti/tcti-device.c]: Fixed

**Comment by tstruk** [comment]
@flihp Thanks for your feedback. I have reworked the code accordingly, updated the commit message, and added unit tests to cover the poll() interface. Please have a look.
